### ISSUE TITLE
[BUG] Cannot add data filter when creating monitor in alerting

### ISSUE THREAD

**Issue by kavilla** [opensearch-project/alerting-dashboards-plugin#558]
@chengleqi commented on [Thu May 25 2023](https://github.com/opensearch-project/OpenSearch-Dashboards/issues/4134)

**Describe the bug**
There is no button to add a filter.

**To Reproduce**
Add monitor, use the visual editor and then add a data filter.

**Expected behavior**
There is a button that allows me to add a data filter.

**OpenSearch Version**
2.7.0

**Dashboards Version**
2.7.0

**Plugins**
![image](https://github.com/opensearch-project/OpenSearch-Dashboards/assets/53958238/79f704b1-074a-4198-b4fe-672fefec5a49)

**Screenshots**
![1685005580959](https://github.com/opensearch-project/OpenSearch-Dashboards/assets/53958238/c8ec2641-cf71-48d5-a30f-cdbedd5b393c)

**Host/Environment (please complete the following information):**
 - OS: Use docker compose commad method in quick start
 - Browser and version: Google Chrome/113.0.5672.127

**Additional context**

**Comment by chengleqi** [comment]
mark

**Comment by AWSHurneyt** [comment]
@chengleqi Could you clarify the issue you're experiencing? In the second screenshot, it looks like the UI component to add a data filter is present. Is that component not appearing under certain circumstances for you?

**Comment by chengleqi** [comment]
Hi @AWSHurneyt , I am in the same situation as this issue #545 , where I don't have a button to add a data filter.
![image](https://github.com/opensearch-project/alerting-dashboards-plugin/assets/53958238/e06112df-f037-481f-bd92-6a1c9e290c16)

**Comment by chengleqi** [comment]
When I click outside this popup, it doesn't add the data filter that I have configured, so I believe there should be a button in this area.

**Comment by thomass4t** [comment]
Short question:
Will it be pushed to the apk packages in version 2.8?
https://artifacts.opensearch.org/releases/bundle/opensearch/2.x/apt 
I don't know about the release or update cycles.

**Comment by AWSHurneyt** [comment]
Hi @thomass4t,
Until we make a patch release, such as 2.8.1, this fix would not update the current release artifacts. We don't have a scheduled date for 2.8.1 at the moment. This fix is present in OpennSearch 2.9.
https://opensearch.org/releases.html

As a workaround, data filters can be configured using the `extraction query editor` UI. Here's an example of what it would look like if you wanted to add `products.product_id is 123` and `user is steph` filters to the `opensearch_dashboards_sample_data_ecommerce` index.
```
{
  "size": 0,
  "aggregations": {
    "composite_agg": {
      "composite": {
        "sources": [
          {
            "user": {
              "terms": {
                "field": "user"
              }
            }
          }
        ]
      },
      "aggs": {}
    }
  },
  "query": {
    "bool": {
      "filter": [
        {
          "range": {
            "order_date": {
              "gte": "{{period_end}}||-1h",
              "lte": "{{period_end}}",
              "format": "epoch_millis"
            }
          }
        },
        {
          "term": {
            "products.product_id": 123
          }
        },
        {
          "term": {
            "user": "steph"
          }
        }
      ]
    }
  }
}
```

**Comment by AWSHurneyt** [comment]
Closing this issue as the fix in PR https://github.com/opensearch-project/alerting-dashboards-plugin/pull/574 has been merged and released with 2.9. Please feel free to reopen this issue if this issue persists.

**Comment by thomass4t** [comment]
@AWSHurneyt 
Thanks for the reply and the description about the workaround 👍
